doc/spec/package-lock.md

```diff
@@ -0,0 +1,273 @@
+# package-lock and npm-shrinkwrap
+
+`npm` can have one of two different lock files:
+
+* `package-lock.json`, which is ordinarily always present and is never published.
+* `npm-shrinkwrap.json`, which is created with `npm shrinkwrap` and usually published.
+
+You can only have one of them and in the event that you have both,
```

Comment:
@jsdnxx - They mean different things though. "You can" implies that it's a requirement, whereas "you should" implies that it's a good idea but not mandatory. I guess it depends on how strong the suggestion/requirement should be.